src/main/kotlin/app/revanced/patcher/fingerprint/method/impl/MethodFingerprint.kt

```diff
@@ -54,6 +202,30 @@ abstract class MethodFingerprint(
                         break@classes // if the resolution succeeded, continue with the next fingerprint
         }
 
+        /**
+         * Resolve using map built in [createMethodLookupMap]
+         */
+        internal fun MethodFingerprint.resolveUsingLookupMap(context: BytecodeContext, logger : Logger = NopLogger): Boolean {
```

Comment:
Are the `resolveUsingLookupMap` extension functions even used anywhere in this project? They include a logger parameter but other resolve methods don't, this should be consistently applied to all or no extension function here.

Reply:
> I think the logger should be a field that's set by another class (ie Patcher),

Currently, we use extension functions instead of simply adding the functions right to the `MethodFingerprint` class. Eventually, this will happen by a refactor, and since the functions will not be static, a logger can only be injected as a parameter, or it has to be static somewhere, but the latter is not elegant. For now, removing the logger completely should be fine, as logging in itself is worth a PR.